### PR TITLE
Formmating: Add white space around vertical bar

### PR DESCRIPTION
**Pull request by eksperimental**
It adds a white space around the vertical bar, improving readability of the code.
Carefully edited, and thoroughly reviewed three times.

While the amount of lines edited is significant, I think it's a great improvement in the clarity of the code.

### CODE REVIEW

**Comment by josevalim** [comment]
@lexmag it is your call. :)

**Comment by ericmj** [comment]
I have no strong opinion on it but personally I am not convinced it's an improvement.

**Comment by whatyouhide** [comment]
It's a huge amount of changes and we'll be blaming @eksperimental for years in git :P, but maybe this is an improvement after all (consistency is good). We'd have to pay a lot of attention when reviewing PRs so that we don't reintroduce `x|y` though.

File: lib/elixir/lib/enum.ex [hunk 1]
```diff
@@ -943,8 +943,8 @@ defmodule Enum do
       end) |> :lists.reverse()
 
     case list do
-      []    -> []
-      [_|t] -> t  # Head is a superfluous intersperser element
+      []      -> []
```
Comment (lexmag) [inline lib/elixir/lib/enum.ex]: Let's fix this the other way around to `[] -> []`.

File: lib/elixir/lib/enum.ex [hunk 2]
```diff
@@ -2784,7 +2784,7 @@ defimpl Enumerable, for: List do
   def reduce(_,     {:halt, acc}, _fun),   do: {:halted, acc}
   def reduce(list,  {:suspend, acc}, fun), do: {:suspended, acc, &reduce(list, &1, fun)}
   def reduce([],    {:cont, acc}, _fun),   do: {:done, acc}
-  def reduce([h|t], {:cont, acc}, fun),    do: reduce(t, fun.(h, acc), fun)
+  def reduce([h | t], {:cont, acc}, fun),    do: reduce(t, fun.(h, acc), fun)
```
Comment (lexmag) [inline lib/elixir/lib/enum.ex]: There is a broken `do:`.

File: lib/elixir/lib/enum.ex [hunk 3]
```diff
@@ -2800,7 +2800,7 @@ defimpl Enumerable, for: Map do
   defp do_reduce(_,     {:halt, acc}, _fun),   do: {:halted, acc}
   defp do_reduce(list,  {:suspend, acc}, fun), do: {:suspended, acc, &do_reduce(list, &1, fun)}
   defp do_reduce([],    {:cont, acc}, _fun),   do: {:done, acc}
-  defp do_reduce([h|t], {:cont, acc}, fun),    do: do_reduce(t, fun.(h, acc), fun)
+  defp do_reduce([h | t], {:cont, acc}, fun),    do: do_reduce(t, fun.(h, acc), fun)
```
Comment (lexmag) [inline lib/elixir/lib/enum.ex]: There is a broken `do:`.

File: lib/elixir/lib/enum.ex [hunk 4]
```diff
@@ -109,7 +109,7 @@ defprotocol Enumerable do
       def reduce(_,     {:halt, acc}, _fun),   do: {:halted, acc}
       def reduce(list,  {:suspend, acc}, fun), do: {:suspended, acc, &reduce(list, &1, fun)}
       def reduce([],    {:cont, acc}, _fun),   do: {:done, acc}
-      def reduce([h|t], {:cont, acc}, fun),    do: reduce(t, fun.(h, acc), fun)
+      def reduce([h | t], {:cont, acc}, fun),    do: reduce(t, fun.(h, acc), fun)
```
Comment (lexmag) [inline lib/elixir/lib/enum.ex]: There is a broken `do:` indentation.

File: lib/elixir/lib/enum.ex [hunk 5]
```diff
@@ -2467,13 +2467,13 @@ defmodule Enum do
 
   ## fetch
 
-  defp do_fetch([h|_], 0), do: {:ok, h}
-  defp do_fetch([_|t], n), do: do_fetch(t, n - 1)
+  defp do_fetch([h | _], 0), do: {:ok, h}
+  defp do_fetch([_ | t], n), do: do_fetch(t, n - 1)
   defp do_fetch([], _),    do: :error
```
Comment (lexmag) [inline lib/elixir/lib/enum.ex]: There is a broken `do:` indentation.

File: lib/elixir/lib/float.ex [hunk 6]
```diff
@@ -250,8 +250,8 @@ defmodule Float do
     :erlang.float_to_binary(float, expand_compact(options))
   end
 
-  defp expand_compact([{:compact, false}|t]), do: expand_compact(t)
-  defp expand_compact([{:compact, true}|t]),  do: [:compact|expand_compact(t)]
-  defp expand_compact([h|t]),                 do: [h|expand_compact(t)]
+  defp expand_compact([{:compact, false} | t]), do: expand_compact(t)
+  defp expand_compact([{:compact, true} | t]),  do: [:compact | expand_compact(t)]
+  defp expand_compact([h | t]),                 do: [h | expand_compact(t)]
   defp expand_compact([]),                    do: []
```
Comment (lexmag) [inline lib/elixir/lib/float.ex]: There is a broken `do:` indentation.

File: lib/elixir/lib/hash_set.ex [hunk 7]
```diff
@@ -174,7 +174,7 @@ defmodule HashSet do
 
   defp do_fold_each([], acc, _fun),   do: acc
   defp do_fold_each([t], acc, fun),   do: fun.(t, acc)
-  defp do_fold_each([t|n], acc, fun), do: do_fold(n, fun.(t, acc), fun, @node_size)
+  defp do_fold_each([t | n], acc, fun), do: do_fold(n, fun.(t, acc), fun, @node_size)
```
Comment (lexmag) [inline lib/elixir/lib/hash_set.ex]: There is a broken do: indentation.

**Comment by eksperimental** [comment]
@lexmag  all fixed, and  a few more than i found with a search command. I looked for changes in line containing `->` but i overlooked the ones with `do: `, so thanks for that!

the CI build failed, but I think it's unrelated, since locally it just passed

**Comment by vikram7** [comment]
I think this is a pretty good improvement for readability. Of the handful of Elixir docs that I've read through (like [here](http://elixir-lang.org/docs/stable/elixir/List.html#flatten/1)), `[one | two]` seems to be more common than `[one|two]`. If that's the case, then it would be worth making sure they match up. :smile:

**Comment by eksperimental** [comment]
@lexmag: rebased to master

**Comment by lexmag** [comment]
@eksperimental 💛

**Comment by lexmag** [comment]
Seems GitHub has fixed the whitespace ignoring feature `?w=1`, with it we have 0 files changed. 😄

**Comment by eksperimental** [comment]
@lexmag cool! 
thanks for merging.. I didn't know about [that cool feature](https://github.com/elixir-lang/elixir/pull/4507/files?w=1)

**Pull request merged**